# Exploratory Analysis
This notebook will look into the initial analysis to see what data is held in the Met Office website. We will find the attributes that are collected/shown to the public and get a feel for the dataset.

In [1]:
# packages
from clean import create_table
from location import Location
from location_attr import LocationAttributes

## Choose a location
First we will choose a location. If you would like to know what locations there are to choose from, either look at location.py or start writing "Location."and press "ctrl + space" which will show you a list of all weather stations around the UK.

In [2]:
# location variables
LOCATION = Location.HEATHROW
location_attributes = LocationAttributes(LOCATION)

# data frame
location_attributes.df = create_table(
    location_name=LOCATION.value,
    location_url=location_attributes.url,
    data_refresh=False,
)

## Explore the data
Now we have a clean dataset from a chosen location, we can see what data is in it.

In [3]:
# What are the attributes of this dataset
location_attributes.df.columns

Index(['year', 'month', 'tmax_degC', 'tmin_degC', 'af_days', 'rain_mm',
       'sun_hours', 'is_predicted', 'decade', 'worded_month'],
      dtype='object')

## The columns
As we can see above, here are the columns in this dataset and what each of these represent.

- **year** - An integer that represents the year
- **decade** - An integer that represents the decade (analyst created in `clean.py`)
- **month** - An integer that represents the month
- **worded_month** - A string that represents the month (analyst created in `clean.py`)
- **tmax_degC** - A float (a number with decimals) that represents the maximum temperature that day in degrees celcius
- **tmin_degC** - A float (a number with decimals) that represents the minimum temperature that day in degrees celcius
- **af_days** - A float (a number with decimals) that represents the number of days of air frost
- **rain_mm** - A float that represents the amount of rain in the area in millimetres
- **sun_hours** - A float that represents the amount of sun in hours
- **is_predicted** - A boolean (True/False) to show whether the record is predictive or an observation.

### The attributes and the measurements
We can see that every month and year, we get measurements for temperature, rain, air frost and sun.

This means the granularity of this dataset is per month. We can see measurements for each month from the opening of the weather station.

In [4]:
location_attributes.opening_year

np.int64(1948)

In [5]:
df = location_attributes.df

df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 922 entries, 0 to 921
Data columns (total 10 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   year          922 non-null    Int64  
 1   month         922 non-null    Int64  
 2   tmax_degC     922 non-null    Float64
 3   tmin_degC     922 non-null    Float64
 4   af_days       910 non-null    Float64
 5   rain_mm       922 non-null    Float64
 6   sun_hours     814 non-null    Float64
 7   is_predicted  922 non-null    object 
 8   decade        922 non-null    int64  
 9   worded_month  922 non-null    object 
dtypes: Float64(5), Int64(2), int64(1), object(2)
memory usage: 78.5+ KB


In [6]:
df.describe()

,year,month,tmax_degC,tmin_degC,af_days,rain_mm,sun_hours,decade
count,922.0,922.0,922.0,922.0,910.0,922.0,814.0,922.000000
mean,1985.91757,6.489154,15.028633,7.171475,2.891209,50.90282,128.965479,1981.475054
std,22.191588,3.449715,5.804454,4.466427,4.599018,29.888276,64.694544,22.474353
min,1948.0,1.0,0.8,-4.6,0.0,0.3,18.5,1940.000000
25%,1967.0,3.25,9.9,3.4,0.0,28.25,69.45,1960.000000
50%,1986.0,6.0,14.9,6.8,0.0,47.4,125.85,1980.000000
75%,2005.0,9.0,20.175,11.2,4.0,69.45,176.05,2000.000000
max,2024.0,12.0,28.3,16.7,28.0,174.8,310.1,2020.000000


## Looking into numeric columns
We see that ther are up to 922 records since the weather station has been opened. We see that the `sun (hours)` has less which could be because it has more nulls which might be because it didn't have the capabilities to record the number of days had sun for the first part of the weather stations longevity.


## Overview
We see that the dataset is large, with 922 records spanning 8 decades. This means that it can be analysed well going forward.

**Questions that could be answered are:**
- What are the trends through the decades?